api/types/generatorargs.go

```diff
@@ -13,6 +13,9 @@ type GeneratorArgs struct {
 	// NamePrefix + this.Name + hash(content of generated resource).
 	Name string `json:"name,omitempty" yaml:"name,omitempty"`
 
+	// Mutability of the generated resource.
+	Immutable *bool `json:"immutable,omitempty" yaml:"immutable,omitempty"`
```

Comment:
Why make this a ptr to bool instead of just bool?

Reply:
Because if it becomes possible to set default mutability at a higher level (e.g. namespace, cluster), it would therefore be applied if mutability is not specified at resource level.

Reply:
The behavior of defaulting a field is also possible with a custom mutating webhook.